### PR TITLE
licensing: create dashboard page for Cody enterprise

### LINKED ISSUE #59949: Implement Cody Dashboard

### PR DESCRIPTION
**Pull request by BolajiOlajide**
Closes #59949

Currently, we don't have a dashboard page to show enterprise users more information about Cody and how to use it.

## Test plan

On an non-dotcom instance, navigating to `/cody` will take you to an enterprise Dashboard page.

CleanShot 2024-02-07 at 16 58 12

CleanShot 2024-02-07 at 16 57 56

### CODE REVIEW

File: client/web/src/routes.tsx [hunk 1]
```diff
@@ -169,24 +170,9 @@ const codyRoutes: readonly RouteObject[] = [
         path: PageRoutes.CodySearch,
         element: <LegacyRoute render={props => <CodySearchPage {...props} />} />,
     },
-    // TODO: [TEMPORARY] remove this redirect route when the marketing page is added.
     {
-        path: `${PageRoutes.Cody}/*`,
```
Comment (BolajiOlajide) [inline client/web/src/routes.tsx:174]: @sqs Just confirming that it's okay to remove this since we have the marketing page added now.

**Review by taiyab — CHANGES_REQUESTED** [review]
In the preview it looks like you have Code Search enabled but the Code Search banner is still appearing (it shouldn't).

This banner should only appear when they don't have a Code Search license.

**Comment by BolajiOlajide** [comment]
> In the preview it looks like you have Code Search enabled but the Code Search banner is still appearing (it shouldn't).
> 
> This banner should only appear when they don't have a Code Search license.

Done.

**Review by taiyab — COMMENTED** [review]
All good! 👍

**Review by vovakulikov — APPROVED** [review]

**Review by taiyab — APPROVED** [review]
Approved.

**Pull request merged**